zapcore/field.go

```diff
@@ -208,7 +208,12 @@ func addFields(enc ObjectEncoder, fields []Field) {
 func encodeStringer(key string, stringer interface{}, enc ObjectEncoder) (err error) {
 	defer func() {
 		if v := recover(); v != nil {
-			err = fmt.Errorf("PANIC=%v", v)
+			val := reflect.ValueOf(stringer)
```

Comment:
This block looks very similar to the logic in the fmt package uses to determine if `nil` should be used,
https://golang.org/src/fmt/print.go#L540

Should we reference this?